app/lib/client.server.ts

```diff
@@ -0,0 +1,179 @@
+import { createRecord, upsertRecord, deleteRecord, getChangeStatus } from './dns.server';
+import type { RecordType } from './dns.server';
+import {
+  createRecordInDB,
+  updateRecordInDB,
+  deleteRecordInDB,
+  findRecordById,
+  doesRecordExistInDB,
+} from '../models/record.server';
+import { RecordStatus } from '@prisma/client';
+
+export const createUserDomain = async (
+  hostedZoneId: string,
+  username: string,
+  type: RecordType,
+  name: string,
+  value: string
+) => {
+  const changeId = await createRecord(hostedZoneId, type, name, value);
+
+  if (!changeId) {
+    throw new Error('Could not get ChangeID from creating a record in Route 53');
+  }
+
+  const response = await getChangeStatus(changeId);
+
+  if (!response) {
+    throw new Error('Could not get ChangeStatus from Route 53');
+  }
+
+  let status: RecordStatus;
+  switch (response) {
+    case 'PENDING':
+      status = RecordStatus.pending;
+      break;
+    case 'INSYNC':
+      status = RecordStatus.active;
+      break;
+    default:
+      status = RecordStatus.error;
+      break;
+  }
+
+  const existingRecords = await doesRecordExistInDB(type, name, value);
+
+  if (existingRecords) {
+    throw new Error('Record already exists');
+  }
+
+  const result = await createRecordInDB(
+    username,
+    type,
+    name,
+    value,
+    new Date(),
+    new Date(),
+    new Date(new Date().setMonth(new Date().getMonth() + 6)),
```

Comment:
Make a function to better document this:

```ts
const dateFromNow = (months: number) => new Date(new Date().setMonth(new Date().getMonth() + months));
```

Reply:
It is not a good looking code either